src/main/java/net/neoforged/neoforge/client/entity/animation/json/JsonAnimationParser.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.entity.animation.json;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import it.unimi.dsi.fastutil.Pair;
+import java.util.Map;
+import net.minecraft.client.animation.AnimationChannel;
+import net.minecraft.client.animation.AnimationDefinition;
+import net.minecraft.client.animation.Keyframe;
+import net.minecraft.client.animation.KeyframeAnimations;
+import net.minecraft.util.GsonHelper;
+import org.joml.Vector3f;
+
+/**
+ * A parser for parsing JSON entity animation files.
+ */
+public final class JsonAnimationParser {
```

Comment:
Is there a reason not to use a codec here?

Reply:
No major reason, but I will point out that a lot of the resource pack stuff does not use codecs. Additionally, with the new registry stuff, I think it would be best to not use a codec, since that could introduce mutability into a codec.

Reply:
I'm confused. How would the type map be mutable during reload? We would only be allowing registering to it at a certain point in time, and the types wouldn't change. The definitions would change, yes, but the codec is constructing the value there, so it's independent.

Reply:
Anyway, unless you or someone else really wants a codec, I don't think it's worth putting effort into making one.